src/sinks/azure_monitor_logs.rs

```diff
@@ -32,25 +33,59 @@ fn default_host() -> String {
     "ods.opinsights.azure.com".into()
 }
 
-#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+/// Configuration for the `azure_monitor_logs` sink.
+#[configurable_component]
+#[derive(Clone, Debug, Default)]
 #[serde(deny_unknown_fields)]
 pub struct AzureMonitorLogsConfig {
+    /// The [unique identifier][uniq_id] for the Log Analytics workspace.
+    ///
+    /// [uniq_id]: https://docs.microsoft.com/en-us/azure/azure-monitor/platform/data-collector-api#request-uri-parameters
     pub customer_id: String,
+
+    /// The [primary or the secondary key][shared_key] for the Log Analytics workspace.
+    ///
+    /// [shared_key]: https://docs.microsoft.com/en-us/azure/azure-monitor/platform/data-collector-api#authorization
     pub shared_key: String,
+
+    /// The [record type][record_type] of the data that is being submitted.
+    ///
+    /// Can only contain letters, numbers, and underscores (_), and may not exceed 100 characters.
+    ///
+    /// [record_type]: https://docs.microsoft.com/en-us/azure/azure-monitor/platform/data-collector-api#request-headers
+    #[configurable(validation(pattern = "[a-zA-Z0-9_]{1,100}"))]
```

Comment:
YES! I love this. ❤️

Reply:
Yeah! It'll be so much nicer to start baking validation in to the struct definition.

I also have plans to make the macro compile any `pattern` validators with the `regex` crate to tell the user if the regular expression itself is actually valid or not. 😈